src/main.rs

```diff
@@ -3964,4 +3998,127 @@ mod tests {
         let var_decl = model.vars.get(0).unwrap();
         var_decl.init == "Iso8601Date";
     }
+
+    #[test]
+    fn type_directive_imported() {
+        let team_id = "
+            // @typescript-to-gdscript-type: int
+            export type TeamId = number;
+        ";
+        let src = "
+            import { TeamId } from \"./team-id.ts\";
+            export interface A { interface_value: TeamId; }
+        ";
+        let models = parse_and_get_models_with_imports(
+            "type-directive-imported.ts",
+            &src,
+            &[&parse_from_string("team-id.ts", team_id)],
+        );
+        let interfaceValue = &models[0].var_descriptors[0];
+        assert_eq!(interfaceValue.name, "interface_value");
+        assert_eq!(interfaceValue.decl_type, Some(String::from("int")));
+    }
+
+    #[test]
+    fn type_directive_local() {
+        let src = "
+            // @typescript-to-gdscript-type: int
+            export type TeamId = number;
+            export interface A { interface_value: TeamId; }
+        ";
+        let models = parse_and_get_models("type-directive-across-modules.ts", &src);
+        let interfaceValue = &models[0].var_descriptors[0];
+        assert_eq!(interfaceValue.name, "interface_value");
+        assert_eq!(interfaceValue.decl_type, Some(String::from("int")));
+    }
+    #[test]
+    fn type_directive_on_property() {
+        let src = "
+            export type TeamId = number;
+            export interface A { 
+                // @typescript-to-gdscript-type: unknown
+                interface_value: TeamId; 
+            }
+        ";
+        let models = parse_and_get_models("type-directive-on-property.ts", &src);
+        let interfaceValue = &models[0].var_descriptors[0];
+        assert_eq!(interfaceValue.name, "interface_value");
+        assert_eq!(interfaceValue.decl_type, Some(String::from("unknown")));
+    }
+    #[test]
+    #[ignore]
+    fn type_directive_override_property() {
```

Comment:
This test also doesn't work right now, but it can be easily worked around by only using the directive on the type. (I'm assuming we don't need to do this). This does work across modules though.